### DISCUSSION TITLE
Building an app for Android using `tauri`

### DISCUSSION THREAD

**Discussion by Raj2032** [tauri-apps/tauri#3116] [General]
![image](https://user-images.githubusercontent.com/29293830/147403760-cf2fc624-f39a-4043-92be-1cc2d2d60b40.png)

Hey guys,

I am looking to make a web based application for Android which can scan qr codes, image process it using Rust.

I know that `tauri` as of now does not support Android but it does support web and webview is supported by Android.

The qr code would be a third party library I would use into my application, something like `rqrr`

https://github.com/WanzenBug/rqrr

If I compile the entire project including the `rqrr` library into a web application, would it just work out of the box or do i have to drastically change something?

There are two libraries I looked at which are Rocket UI:

https://rocket.rs

and Tauri:

https://tauri.studio/en/

Would they be able to acheive what I am looking for? And if so which one should I use in terms of ease of use, animations, and see the screenshot how there is a camera view in the middle of the screen and the rest is the app, would that be also supported?

**Comment by FabianLars** [comment]
Both libs/frameworks you linked are **not** UI libs!
Rocket is a server-side framework, for example for creating REST apis and stuff like that.
Tauri is a framework to create desktop **apps** (and soon-ish mobile **apps**), but it needs you to bring your own UI like react etc. ("tauri doesn't care about your frontend").

> I know that tauri as of now does not support Android but it does support web and webview is supported by Android.
> If I compile the entire project including the rqrr library into a web application

Without tauri actually supporting Android, you can't build a tauri app for android. you can't compile tauri apps **into** web apps (html,js,css,wasm), it builds executable binaries (and installers for it), which **includes** the web app stuff you brought yourself like mentioned above.

Edit: i guess you're probably looking for things like yew, seed, dominator?

**Reply by Raj2032** [reply]
I see mate, thanks.

> Edit: i guess you're probably looking for things like yew, seed, dominator?

I have never heard of these but yeah I can look into those.
